### DISCUSSION TITLE
'WIP' status of C++ implementation?

### DISCUSSION THREAD

**Discussion by gavanderhoorn** [rpiRobotics/linear-subproblem-solutions#30] [Q&A]
The README currently says this about the C++ implementation:

https://github.com/rpiRobotics/ik-geo/blob/4e5a0f6a9d40f96f7c89ce1c86113350eea809d5/README.md?plain=1#L72

could you perhaps clarify what the status of it is?

Would the Rust implementation be more complete / a better entry point to IK-Geo?

**Comment by aelias36** [comment]
C++ implementation is ready for use and we already have other people using it, but we haven't completing timing tests yet.
C++, Rust, and MATLAB are all good entry points.

Thanks for your interest!

**Reply by emiladaxis** [reply]
Hello @aelias36! First of all, great work on making the C++ implementation ready for use. 

The kinematic notation for the Fanuc CRX has me somewhat confused. Are any of the C++ IK implementations suitable for this robot, e.g. IK_3_parallel_2_intersecting? If I am not mistaken it has three pair of intersecting axes.

**Reply by aelias36** [reply]
Hi @emiladaxis,

You are correct that the CRX has 3 pairs of intersecting axes. This requires a search-based solutions, and the IK function for robots with just 2 intersecting axes should work. However, this is not the most efficient way to perform inverse kinematics since the extra intersecting axes means that we don't have to use Subproblem 5. 

I implemented the following IK function in MATLAB, but it hasn't been ported to the other languages yet:
https://github.com/rpiRobotics/subproblem-polynomial/blob/main/IK_3_pairs_intersecting.m
